### PR TITLE
Pz/schames 008 - validate single voucher

### PR DESCRIPTION
**Pull request by p-zielinski**
## 20231023

**Added schemas**
- VouchersValidateRequestBody
- VouchersValidateDiscountRequestBody
- VouchersValidateGiftRequestBody
- VouchersValidateLoyaltyRequestBody
- Referrer
- OrderItem
- Session
- VouchersValidateValidResponseBody
- VouchersValidateInvalidResponseBody
- OrderRedemptions
- SimpleProductDiscountUnit
- SimpleSkuDiscountUnit
- DiscountUnit
- DiscountUnitMultiple
- DiscountUnitMultipleOneUnit
- DiscountAmount
- DiscountPercent
- DiscountFixed
- DiscountAmountVouchersEffectTypes
- DiscountPercentVouchersEffectTypes
- DiscountFixedVouchersEffectTypes
- DiscountUnitVouchersEffectTypes
- DiscountVouchersEffectTypes
- DiscountVouchersTypes
- InapplicableToResultList
- ApplicableToResultList
- InapplicableTo
- ApplicableTo
- ApplicableToEffect
- VouchersValidateResponseBody

**Removed schemas**
- 6_req_validate_voucher

**Endpoints changes**
- `/v1/vouchers/{code}/validate`
  - Request parameters schema was replaced with new one: `VouchersValidateRequestBody` (old one: `6_req_validate_voucher` *has been deleted*)
  - Response schema was replaced with new one: `VouchersValidateResponseBody` (old one: `anyOf`: `6_res_validate_voucher`, `6_res_validate_voucher_false`)
- `/v1/validation-object`
  - Response schema was replaced with new one: `VouchersValidateResponseBody` (old one: `anyOf`: `6_res_validate_voucher`, `6_res_validate_voucher_false`)

Preview:
https://docs.voucherify.io/v2018-08-01-piotr-502/reference/validate-voucher

### CODE REVIEW

File: reference/OpenAPI.json [hunk 1]
```diff
@@ -381,6 +381,1013 @@
...
+          "product": {
+            "$ref": "#/components/schemas/SimpleProductDiscountUnit",
+            "description": "Contains information about the product."
+          },
+          "sku": {
+            "$ref": "#/components/schemas/SimpleSkuDiscountUnit"
+          }
+        }
+      },
+      "DiscountUnitMultiple": {
```
Comment (weronika-kurczyna) [inline reference/OpenAPI.json:1174]: image
- `type` was not returned here (inside the object in array)
- please verify if the `unit_type` is required
image
it looks like a duplicate - in `UNIT` we have `Simple SKU Discount Unit Object`

File: reference/OpenAPI.json [hunk 2]
```diff
@@ -381,6 +381,1013 @@
...
+            "description": "Contains information about the sku."
+          }
+        },
+        "required": [
+          "type",
+          "unit_off",
+          "effect"
+        ]
+      },
+      "DiscountAmount": {
```
Comment (weronika-kurczyna) [inline reference/OpenAPI.json:1253]: The other types for `discount` in response (https://docs.voucherify.io/v2018-08-01-piotr-502/reference/validate-voucher) don't have required properties (expect the UNIT_MULTIPLE). I'm not sure if all the fields are required :)

**Review by weronika-kurczyna — CHANGES_REQUESTED** [review]
As we discussed, we need to decide whether we want to separate the fields for body (old version) or to define all the same place.
For example, if you want redeem code from loyalty card the request body requires `reward` etc.

**Comment by weronika-kurczyna** [comment]
+ conflicts

File: reference/OpenAPI.json (hunk 1 above)
Comment (p-zielinski) [reply reference/OpenAPI.json:1174]: you are right, i messed up :)

**Review by weronika-kurczyna — APPROVED** [review]
image
I try not to have texts in the documentation that are not formatted correctly.
Besides that, everything's fine, it looks much better when we have a division in the body params👍🏼

**Pull request merged**